### PR TITLE
 Tradheli: Correct filter param descriptions in AC_AttitudeControl library

### PR DESCRIPTION
**Pull request by bnsgeyer**
This remove the old filter param description from AC 3.6 and adds the new filter param descriptions for AC 4.0

### CODE REVIEW

File: libraries/AC_AttitudeControl/AC_AttitudeControl_Heli.cpp [hunk 1]
```diff
@@ -135,12 +169,29 @@ const AP_Param::GroupInfo AC_AttitudeControl_Heli::var_info[] = {
...
+    // @Range: 1 50
+    // @Increment: 1
     // @Units: Hz
-    // @Range: 1 20
+    // @User: Standard
+
+    // @Param: RAT_YAW_FLTE
+    // @DisplayName: Yaw axis rate controller error frequency in Hz
+    // @Description: Yaw axis rate controller error frequency in Hz
+    // @Range: 1 50
```
Comment (rmackay9) [inline libraries/AC_AttitudeControl/AC_AttitudeControl_Heli.cpp]: RAT_YAW_FLTE upper range is 10 in multicopters..

File: libraries/AC_AttitudeControl/AC_AttitudeControl_Heli.cpp [hunk 2]
```diff
@@ -135,12 +169,29 @@ const AP_Param::GroupInfo AC_AttitudeControl_Heli::var_info[] = {
...
     // @Increment: 1
+    // @Units: Hz
+    // @User: Standard
+
+    // @Param: RAT_YAW_FLTD
+    // @DisplayName: Yaw axis rate controller derivative frequency in Hz
+    // @Description: Yaw axis rate controller derivative frequency in Hz
+    // @Range: 1 50
```
Comment (rmackay9) [inline libraries/AC_AttitudeControl/AC_AttitudeControl_Heli.cpp]: RAT_YAW_FLTD upper range is 20 in multicopters..

File: libraries/AC_AttitudeControl/AC_AttitudeControl_Heli.cpp [hunk 3]
```diff
@@ -92,12 +109,29 @@ const AP_Param::GroupInfo AC_AttitudeControl_Heli::var_info[] = {
...
+    // @Increment: 1
     // @Units: Hz
-    // @Range: 1 20
+    // @User: Standard
+
+    // @Param: RAT_PIT_FLTE
+    // @DisplayName: Pitch axis rate controller error frequency in Hz
+    // @Description: Pitch axis rate controller error frequency in Hz
+    // @Range: 1 50
```
Comment (rmackay9) [inline libraries/AC_AttitudeControl/AC_AttitudeControl_Heli.cpp]: RAT_PIT_FLTE and _FLTD upper range is 100 in multicopters..

File: libraries/AC_AttitudeControl/AC_AttitudeControl_Heli.cpp [hunk 4]
```diff
@@ -49,12 +49,29 @@ const AP_Param::GroupInfo AC_AttitudeControl_Heli::var_info[] = {
...
+    // @Increment: 1
     // @Units: Hz
-    // @Range: 1 20
+    // @User: Standard
+
+    // @Param: RAT_RLL_FLTE
+    // @DisplayName: Roll axis rate controller error frequency in Hz
+    // @Description: Roll axis rate controller error frequency in Hz
+    // @Range: 1 50
```
Comment (rmackay9) [inline libraries/AC_AttitudeControl/AC_AttitudeControl_Heli.cpp]: RAT_RLL_FLTE and _FLTD upper range is 100 in multicopters..

**Comment by rmackay9** [comment]
Looks ok to me but I wonder if the upper range should match multicopters so I've added comments where the two are different.

As a side note, i wonder if we could remove the duplicate param descriptions we see by either leaving them out of the multicopter or tradheli file or if we moved them up to the AC_AttittudeControl.cpp file.  Not something we need to address before this PR goes in though.

**Comment by bnsgeyer** [comment]
@rmackay9 I'm not sure that they should match multicopter.  My philosophy on low pass filters has changed with the incorporation of the Harmonic Notch filter.  Previously these filters were used to remove the noise from the feedback states.  For helicopters, that noise was due to the rotor and was typically at the rotor speed frequency and multiples of it.  With the Harmonic notch and the separate static notch, that rotor noise can be removed or at least attenuated to the point that it is no longer a concern in the feedback states.  So being able to raise the frequency of the low pass filters is huge.  Filters are never a good thing for control of a vehicle and the frequencies that were needed for heli can create a lot of phase loss in the response.  
Sorry for a long reply to a simple question.  I think the heli low pass filter frequencies should be raised as much as possible in all axes as long as the harmonic notch filter is attenuating the rotor noise effectively.  For the yaw axis, I don't think there is any reason that the low pass filters need to be lower than 20 hz.  Heli's are very different from multi's in controlling yaw as heli's have a direct control effector.  As for pitch and roll, I suppose we could make the max limit the same (100 hz).  Again it all depends on how well the notch filter is working.  At that point you might use the low pass filters to attenuate noise from the engine (operates around 16000rpm) or the tail rotor (operates around 3000 rpm).  
So for now I'm not sure that we can make all of these the same as multcopters but maybe some axes like pitch and roll could be the same.

**Comment by rmackay9** [comment]
Great stuff.  Just wanted to be sure that it wasn't a typo.  Merged, thanks!

**Comment by rmackay9** [comment]
FYI, this is included in Copter-4.0.2-rc1

**Pull request merged**